src/Component.js

```diff
@@ -1,52 +1,54 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import copy from 'copy-to-clipboard';
 
+function onClick(event, props) {
+  const {
+    text,
+    onCopy,
+    children,
+    options
+  } = props;
+  const elem = React.Children.only(children);
+  const result = copy(text, options);
+
+  if (onCopy) {
+    onCopy(text, result);
+  }
 
-export const CopyToClipboard = React.createClass({
-  propTypes: {
-    text: React.PropTypes.string.isRequired,
-    children: React.PropTypes.element.isRequired,
-    onCopy: React.PropTypes.func,
-    options: React.PropTypes.shape({
-      debug: React.PropTypes.bool,
-      message: React.PropTypes.string
-    })
-  },
-
-
-  onClick(event) {
-    const {
-      text,
-      onCopy,
-      children,
-      options
-    } = this.props;
-
-    const elem = React.Children.only(children);
-
-    const result = copy(text, options);
-
-    if (onCopy) {
-      onCopy(text, result);
-    }
-
-    // Bypass onClick if it was present
-    if (elem && elem.props && typeof elem.props.onClick === 'function') {
-      elem.props.onClick(event);
-    }
-  },
-
-
-  render() {
-    const {
-      text: _text,
-      onCopy: _onCopy,
-      options: _options,
-      children,
-      ...props
-    } = this.props;
-    const elem = React.Children.only(children);
-
-    return React.cloneElement(elem, {...props, onClick: this.onClick});
+  // Bypass onClick if it was present
+  if (elem && elem.props && typeof elem.props.onClick === 'function') {
+    elem.props.onClick(event);
   }
-});
+}
+
+const CopyToClipboard = (props) => {
+  const {
+    text: _text,
+    onCopy: _onCopy,
+    options: _options,
+    children,
+    ...sanitizedProps
+  } = props;
+  const elem = React.Children.only(children);
+
+  return React.cloneElement(elem, {
+    ...sanitizedProps,
+    onClick: event => onClick(event, props)
```

Comment:
We should not create new function on each render. Please, revert this PR to only change create class and prop types. This PR does too much.